### PR TITLE
Add AppDatabase and LoggedInUser Dao

### PR DESCRIPTION
**Pull request by tiembo**

### CODE REVIEW

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class LoggedInUserDaoTest {
+    private lateinit var database: AppDatabase
+    private lateinit var loggedInUserDao: LoggedInUserDao
+
+    @Before
```
Comment (keyboardsurfer) [inline core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Nit: These annotations can be on the same line as function declaration.

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import android.support.test.InstrumentationRegistry
+import io.plaidapp.core.designernews.data.users.model.LoggedInUser
+import io.plaidapp.test.shared.LiveDataTestUtil
+
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class LoggedInUserDaoTest {
```
Comment (keyboardsurfer) [inline core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Document what this file is testing.

File: core/src/main/java/io/plaidapp/core/data/LoggedInUserDao.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import android.arch.lifecycle.LiveData
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.Query
+import io.plaidapp.core.designernews.data.users.model.LoggedInUser
+
+@Dao
+interface LoggedInUserDao {
```
Comment (keyboardsurfer) [inline core/src/main/java/io/plaidapp/core/data/LoggedInUserDao.kt]: Document

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt (hunk 1 above)
Comment (tiembo) [reply core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Done

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt (hunk 2 above)
Comment (tiembo) [reply core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Done

File: core/src/main/java/io/plaidapp/core/data/LoggedInUserDao.kt (hunk 3 above)
Comment (tiembo) [reply core/src/main/java/io/plaidapp/core/data/LoggedInUserDao.kt]: Done

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        database.close()
+    }
+
+    @Test fun insertAndGetLoggedInUser() {
+        val user = LoggedInUser(
+            id = 1L,
+            displayName = "Loggy L",
+            firstName = "Loggy",
+            lastName = "Loggerson",
+            portraitUrl = "www")
```
Comment (florina-muntenescu) [inline core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: format the file so the closing ")" is on a new line

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        database = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java).build()
+        loggedInUserDao = database.loggedInUserDao()
+    }
+
+    @After fun closeDb() {
+        database.close()
+    }
+
+    @Test fun insertAndGetLoggedInUser() {
+        val user = LoggedInUser(
```
Comment (florina-muntenescu) [inline core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: you can make the user a member of the test

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            id = 1L,
+            displayName = "Loggy L",
+            firstName = "Loggy",
+            lastName = "Loggerson",
+            portraitUrl = "www")
+        loggedInUserDao.insertLoggedInUser(user)
+
+        val userFromDb = LiveDataTestUtil.getValue(loggedInUserDao.getLoggedInUser())
+        assertEquals(user, userFromDb)
```
Comment (florina-muntenescu) [inline core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Can you add comments as we have in other tests?
They should follow the format:
// Given..
// When...
// Then

File: core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt (hunk 4 above)
Comment (tiembo) [reply core/src/androidTest/java/io/plaidapp/core/data/LoggedInUserDaoTest.kt]: Done

**Pull request merged**